### PR TITLE
Rich Text Editor | Add listkeymap extension for Nested list backspace and delete behavior update.

### PR DESCRIPTION
**Pull request by aagash-ni**
# Pull Request

## 🤨 Rationale

- Fix for [Bug 2520998](https://dev.azure.com/ni/DevCentral/_workitems/edit/2520998): Comments | In multiple hierarchy - Backspace to delete the list is not deleting the list.
- Tiptap recently merged this [PR ](https://github.com/ueberdosis/tiptap/pull/4210) in which they introduced the ListKeymap extension which added extra keyhandlers that improve the backspace and the delete keystroke behavior on the list items.

## 👩‍💻 Implementation
- Added [`@tiptap/extension-list-keymap`](https://tiptap.dev/api/extensions/list-keymap) extension to tiptap editor.
- Updated the overall tiptap version from `2.1.6` => `2.1.8` as there was a new patch version available in tiptap package.

## 🧪 Testing

- As this bug involves with multiple key stroke combinations, attempted to implement them in the automated test, But all the browser doesn't seem to behave in a consistent method. Hence manually tested the behavior which fixes the bug mentioned above.
- Tried removing nested list items by pressing both `backspace` as well as `delete` key and verified it to be working properly.

#### Before fix: 
bug

#### After adding ListKeymap extension
afterfix

- Verified major workflows involving backspace and delete keystroke behavior to be as expected.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/rich-text/editor/index.ts
Comment (VikiSekar) [inline packages/nimble-components/src/rich-text/editor/index.ts:1]: await: I am expecting a test case for this PR

Comment (aagash-ni) [reply packages/nimble-components/src/rich-text/editor/index.ts:1]: Updated the description with the rationale for not adding automated cases

File: packages/nimble-components/src/rich-text/specs/README.md [hunk 1]
```diff
@@ -504,6 +504,7 @@ library. For the currently supported features, we will include the following lib
 -   [@tiptap/extension-history](https://www.npmjs.com/package/@tiptap/extension-history)
 -   [@tiptap/extension-italic](https://www.npmjs.com/package/@tiptap/extension-italic)
 -   [@tiptap/extension-list-item](https://www.npmjs.com/package/@tiptap/extension-list-item)
+-   [@tiptap/extension-list-keymap](https://tiptap.dev/api/extensions/list-keymap)
```
Comment (vivinkrishna-ni) [inline packages/nimble-components/src/rich-text/specs/README.md:507]: nit: Can we link the npm site to this package, similar to how other packages are linked? This provides information about the dependency packages like the size, license, etc., 

```suggestion
-   [@tiptap/extension-list-keymap](https://www.npmjs.com/package/@tiptap/extension-list-keymap)
```

**Review by vivinkrishna-ni — APPROVED** [review]

**Review by vikisekarNI — APPROVED** [review]

**Comment by aagash-ni** [comment]
We found a [bug ](https://github.com/ueberdosis/tiptap/issues/4395) in this `listkeymap` extension. Will hold and keep this PR in draft until the issue get fixed in tiptap.

**Comment by atmgrifter00** [comment]
You can bypass me on this PR.

**Comment by rajsite** [comment]
@aagash-ni the changes in this PR are trivial and I have not seen any progress on the bug: https://github.com/ueberdosis/tiptap/issues/4395
I'm going to close the PR, but we can re-open if needed

**Pull request closed without merging**